tests/test_stateful_browser.py

```diff
@@ -445,7 +451,7 @@ def file_get_contents(filename):
 def test_download_link(httpbin):
     """Test downloading the contents of a link to file."""
     browser = mechanicalsoup.StatefulBrowser()
-    browser.open(httpbin.url)
+    open_legacy_httpbin(browser, httpbin)
```

Comment:
Similarly, this could download a page from `httpbin + "/links"`, though it would be text, rather than a PNG.

Reply:
I think downloading a PNG is closer to the typical real-life use-case, so I'd rather stay with the current code.